From bd60e77fa908f290fc6cfefc6a73bda432497653 Mon Sep 17 00:00:00 2001
From: Tiberiu Breana <andrei-tiberiu.breana@nxp.com>
Date: Wed, 8 Mar 2017 12:02:35 +0200
Subject: [PATCH 1578/5242] MLK-14373-1: ARM: dts: Enable Bluetooth interface
 for imx6sl-evk

commit  c05880521d89fbbc9dbbafa38e7fe4db9a6e981a from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable the Bluetooth interface for i.MX 6SoloLite EVK.

Signed-off-by: xlin <xlin@murata.com>
Signed-off-by: Tiberiu Breana <andrei-tiberiu.breana@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sl-evk-btwifi.dts |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sl-evk-btwifi.dts b/arch/arm/boot/dts/imx6sl-evk-btwifi.dts
index 341490d..1c11ae3 100644
--- a/arch/arm/boot/dts/imx6sl-evk-btwifi.dts
+++ b/arch/arm/boot/dts/imx6sl-evk-btwifi.dts
@@ -58,6 +58,22 @@
 		};
 	};
 };
+/* Murata: declare UART4 interface for Bluetooth. */
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4_1>;
+	fsl,uart-has-rtscts;
+	status = "okay";
+	/* for DTE mode, add below change */
+	/* fsl,dte-mode; */
+	/* pinctrl-0 = <&pinctrl_uart4dte_1>; */
+};
+
+&pinctrl_uart4_1 {
+	fsl,pins = <
+		MX6SL_PAD_SD3_DAT3__GPIO5_IO17          0x13069 /* BT_REG_ON */
+	>;
+};
 
 &usdhc1 {
 	pinctrl-names = "default";
-- 
1.7.9.5

